plugin/trino-hive/src/main/java/io/trino/plugin/hive/PartitionsSystemTableProvider.java

```diff
@@ -82,7 +83,7 @@ public PartitionsSystemTableProvider(HivePartitionManager partitionManager, Type
         Table sourceTable = metadata.getMetastore()
                 .getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName())
                 .orElse(null);
-        if (sourceTable == null || isDeltaLakeTable(sourceTable)) {
+        if (sourceTable == null || isDeltaLakeTable(sourceTable) || isIcebergTable(sourceTable)) {
```

Comment:
for follow-up, this returns empty when table not found, while PropertiesSystemTableProvider throws in such case

Reply:
Actually `PropertiesSystemTableProvider` returns `Optional.empty()` as well.
This is covered by the previous test https://github.com/trinodb/trino/pull/10441/commits/3fb545ed8d4c40042966ec147923987ff79208f3#diff-92c37834248d69878a49876bfa440c9f6afbee04b1363d402a04cf7fddc7bb62R2905-R2913